statsmodels/base/model.py

```diff
@@ -751,7 +756,21 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
                 exog = exog[:, None]
             exog = np.atleast_2d(exog)  # needed in count model shape[1]
 
-        return self.model.predict(self.params, exog, *args, **kwargs)
+        predict_results = self.model.predict(self.params, exog, *args, **kwargs)
+
+        if (isinstance(orig_exog, pd.Series) or isinstance(orig_exog, pd.DataFrame) and
```

Comment:
When I checked for the index attribute, it did not uniquely identify the pandas object.  I received the following error:

======================================================================
ERROR: statsmodels.base.tests.test_generic_methods.TestGenericGEEPoisson.test_predict_types
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/brendan/anaconda3/envs/statsmodels-test/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/brendan/statsmodels/statsmodels/base/tests/test_generic_methods.py", line 152, in test_predict_types
    assert_allclose(fitted, res.predict(np.squeeze(p_exog).tolist()),
  File "/home/brendan/statsmodels/statsmodels/base/model.py", line 769, in predict
    return pd.Series(predict_results, index=row_labels)
  File "/home/brendan/anaconda3/envs/statsmodels-test/lib/python3.4/site-packages/pandas/core/series.py", line 151, in __init__
    index = _ensure_index(index)
  File "/home/brendan/anaconda3/envs/statsmodels-test/lib/python3.4/site-packages/pandas/core/index.py", line 3963, in _ensure_index
    return Index(index_like)
  File "/home/brendan/anaconda3/envs/statsmodels-test/lib/python3.4/site-packages/pandas/core/index.py", line 172, in __new__
    subarr = com._asarray_tuplesafe(data, dtype=object)
  File "/home/brendan/anaconda3/envs/statsmodels-test/lib/python3.4/site-packages/pandas/core/common.py", line 2170, in _asarray_tuplesafe
    values = list(values)
TypeError: 'builtin_function_or_method' object is not iterable